pkg/apiserver/api/v1/offering.proto

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+package kubecarrier.api.v1;
+option go_package = "v1";
+
+import "google/api/annotations.proto";
+
+import "types.proto";
+
+message Offering {
+  string name = 1;
```

Comment:
I would like to put name, labels and annotations into our own common ObjectMeta object, similar to Kubernetes.
This common metadata object makes generic handler functions of metadata stuff much easier to implement :)
(And we don't have to add the same fields to EVERY object)

e.g.
```proto
message ObjectMeta {
  string uid = 1; // important to know wheter it's still the same instance we are dealing with or if an object with the same name has been created
  string name = 2;
  google.protobuf.Timestamp created = 3; // can also be creationTimestamp to align with Kubernetes
  google.protobuf.Timestamp deleted = 4; // can also be deletionTimestamp to align with Kubernetes
  string resourceVersion = 5; // super important for concurrency control handling
  map<string, string> labels = 6; // we support label-selectors, so we also need to display them
  map<string, string> annotations = 7; // always nice-to-have as a easy and quick extension point
  uint64 generation = 8; // wether spec/metadata has changed
}
```
this common Metadata should be defined in it's own `meta.proto` and imported here.

In the future we might also want to add and expose `GenerateName` and `Finalizers`.

Other fields should also be put into `Spec` or `Status` sub-objects following the Kubernetes setup.